File: solutions/lua/cars-assemble/1/cars-assemble.lua
local cars = {}

-- returns the amount of working cars produced by the assembly line every hour
function cars.calculate_working_cars_per_hour(production_rate, success_rate)
  return production_rate * success_rate / 100
end

-- returns the amount of working cars produced by the assembly line every minute
function cars.calculate_working_cars_per_minute(production_rate, success_rate)
  return production_rate * success_rate / 100 // 60
end

-- returns the cost of producing the given number of cars
function cars.calculate_cost(cars_count)
  local cheap_cars_number = (cars_count // 10)
  local cheap_cars_cost = cheap_cars_number * 95000
  local expensive_cars_cost = (cars_count - cheap_cars_number * 10) * 10000   
  return cheap_cars_cost + expensive_cars_cost 
end

return cars

File: solutions/clojure/cars-assemble/1/src/cars_assemble.clj
(ns cars-assemble)

(defn production-rate
  "Returns the assembly line's production rate per hour,
   taking into account its success rate"
  [speed]
  (let [success-rate (cond
                       (== speed 0) 0.0
                       (<= speed 4) 1.0
                       (<= speed 8) 0.9
                       (== speed 9) 0.8
                       :else        0.77)] ; Maneja >= 10 y cualquier otro caso
    (* speed 221 success-rate)))

(defn working-items
  "Calculates how many working cars are produced per minute"
  [speed]
  (int (quot (production-rate speed) 60)))
